### PR TITLE
fix bugs when using label smoothing.

### PR DESCRIPTION
**Pull request by whr94621**
in pytorch 0.4 version. Call .dim() on a null tensor will also
 return 1, which will trigger bugs when unsing index_fill_
 afterwards.

### CODE REVIEW

**Comment by whr94621** [comment]
See an example below
``` python
a = torch.tensor([])
print(a.dim())
# return 1
print(a.numel())
# return 0
```

**Comment by da03** [comment]
That's a good point. Thanks!
